### ISSUE TITLE
Create app icons per OS X design guidelines

### ISSUE THREAD

**Issue by cbeams** [bisq-network/bisq#67]
Releated to #66, which deals with creating native installer bundles from the build, it would be good to have proper OS X-style app icons for use in the dock, task switcher, etc.

This would mean adapting the current logo (as attached at 256 and 512px) to follow Apple's [design guidelines for app icons](https://developer.apple.com/library/mac/documentation/UserExperience/Conceptual/AppleHIGuidelines/IconsImages/IconsImages.html).

We can do the same for Windows and Linux afterward, but let's start with OS X.

![logo_256](https://cloud.githubusercontent.com/assets/301810/4433859/d45ecc0e-46f2-11e4-823b-635e5de2d545.png)

![logo_512](https://cloud.githubusercontent.com/assets/301810/4433861/db065216-46f2-11e4-8677-4682bbc3cc23.png)

**Comment by cbeams** [comment]
@raulalgo, is this something that you could pick up?

**Comment by raulalgo** [comment]
Sure! I'm a bit busy these days but I can find some time to do that this week.

**Comment by cbeams** [comment]
Nice! We're planning to publish a very early alpha release this week, complete with native installers (see #66), so it would be great to have proper dock icons, etc. Doesn't have to be perfect at all... just something better than a generic icon :) @ManfredKarrer and I were just talking about this, and he's going to follow up with more information on getting the PSD files etc.

In the meantime, I'd like to assign this issue to you... You should have an invitation to join the bitsquare organization in your email—once you do that, I'll be able to mark you as the assignee. Thanks.

**Comment by ManfredKarrer** [comment]
@raulalgo, @cbeams: I have the psd file for that logo, so I will upload all kind of sized (1024, 512, 256, 128,...) to shared fold (will post it later). 
Basically I think we should stick with that version yet (latest from Marksu with lighter grey, like the above), but we need to check out how it works in different context (webpage, app,. system try icon,...) I have seen already that it does not work so well in the app (light grey seems that it has too less contrast) but lets wait for changes until we see it in the webpage and then finalize it.

**Comment by ManfredKarrer** [comment]
@cbeams: 
I uploaded all icons sizes between 16 and 1024 px including retina versions here: 
http://bitsquare.io/shared/icons/all_icons.zip

**Comment by cbeams** [comment]
Thanks for the upload @ManfredKarrer. Was it intentional to close the issue? Doesn't something special need to be done per the guidelines to use these images as dock and task switcher icons on OS X?

**Comment by ManfredKarrer** [comment]
@cbeams: Ah sorry I thought the task was mainly about providing the images. I have not experience how to get the image icons into the app container so that they are accessed by the OS for dock, task switcher... But I assume that it just need to be in the correct location inside the app container.

**Comment by cbeams** [comment]
> @cbeams: Ah sorry I thought the task was mainly about providing the images.

No worries, sorry for the confusion.

> I have not experience how to get the image icons into the app container so that they are accessed by the OS for dock, task switcher... But I assume that it just need to be in the correct location inside the app container.

That's no problem, actually. The Gradle JavaFX plugin (#66)  already does the work of putting images together into an iconset. The thing that we need here (and that I belive @raulalgo has agreed to do), is to take our current logo and apply the Apple icon design guidelines to to make it look natural in the dock and task switcher. This includes things like applying lighting effects etc.

@raulalgo, this doesn't need to be perfect at all, and the colors in the logo may change further still. It would just be good to have something that looks and feels "right" on OS X for our 0.1.0 release. Something that sends a signal to our alpha testers and screencast viewers that we're serious about building an app that gets native platform integration right :)

**Comment by ManfredKarrer** [comment]
@cbeams: I have not read the details in the guidelines yet, but I assume that the reflection in the dock and the frame in the task switcher are added by the OS. If we should add 3D effects to the Logo itself, I am not sure. Apple was going away from that style (iOS flat design) though on desktop they have not done the transition yet.

**Comment by ManfredKarrer** [comment]
@cbeams:
Here are all icon sizes for packaging with white rounded background.

@raulalgo:
Maybe you could have a look how to improve that a bit?

![shortcut-16](https://cloud.githubusercontent.com/assets/1449498/4473925/571e8036-4959-11e4-974e-dcdb9c0c3b81.png)
![shortcut-16 2x](https://cloud.githubusercontent.com/assets/1449498/4473928/5728bb6e-4959-11e4-8923-6d318d5ff851.png)
![shortcut-32](https://cloud.githubusercontent.com/assets/1449498/4473926/57255d0c-4959-11e4-9827-1c46f10eb079.png)
![shortcut-32 2x](https://cloud.githubusercontent.com/assets/1449498/4473927/5727b4da-4959-11e4-9d80-b2cb539c4f39.png)
![shortcut-64](https://cloud.githubusercontent.com/assets/1449498/4473929/5729525e-4959-11e4-987f-101fb9714c84.png)
![shortcut-64 2x](https://cloud.githubusercontent.com/assets/1449498/4473930/5730b2ce-4959-11e4-8520-7d137a1b7672.png)
![shortcut-128](https://cloud.githubusercontent.com/assets/1449498/4473936/574b50a2-4959-11e4-8f8f-fe78374207b8.png)
![shortcut-128 2x](https://cloud.githubusercontent.com/assets/1449498/4473932/573ee75e-4959-11e4-9010-9f90a20c73f7.png)
![shortcut-256](https://cloud.githubusercontent.com/assets/1449498/4473931/573b4cfc-4959-11e4-8e28-f8510f4c16a1.png)
![shortcut-256 2x](https://cloud.githubusercontent.com/assets/1449498/4473933/57440450-4959-11e4-81b9-c50e9b46333c.png)
![shortcut-512](https://cloud.githubusercontent.com/assets/1449498/4473934/57448718-4959-11e4-9c6c-990fac8c824e.png)
![shortcut-512 2x](https://cloud.githubusercontent.com/assets/1449498/4473935/574981e6-4959-11e4-8811-6163bcab2f57.png)
![shortcut-1024](https://cloud.githubusercontent.com/assets/1449498/4473937/5752c58a-4959-11e4-8e48-c3e6f92b5d89.png)

And here soem screenshot how they look like when deployed:
![bildschirmfoto 2014-10-01 um 12 53 24](https://cloud.githubusercontent.com/assets/1449498/4473952/78c8ed34-4959-11e4-86f2-a03a74f724ea.png)
![bildschirmfoto 2014-10-01 um 12 53 27](https://cloud.githubusercontent.com/assets/1449498/4473951/78c7c9e0-4959-11e4-971f-8f2dd46208c9.png)
![bildschirmfoto 2014-10-01 um 12 53 39](https://cloud.githubusercontent.com/assets/1449498/4473954/78cf8b62-4959-11e4-95d9-d6ce4f5a9441.png)
![bildschirmfoto 2014-10-01 um 12 53 46](https://cloud.githubusercontent.com/assets/1449498/4473953/78ce1f98-4959-11e4-84f1-70a3dc042457.png)

**Comment by cbeams** [comment]
Note, these have been checked in with commit b528256.

**Comment by raulalgo** [comment]
I've prepared three variants to make a bit more solid the icon. One hase the duo tone icon, but I feel it also works nice with blue background and white logo. What do you think?
![icon_512x512_a 2x](https://cloud.githubusercontent.com/assets/8058228/4475479/bd1d3d88-496c-11e4-8293-c6768872e4f3.png)
![icon_512x512_b 2x](https://cloud.githubusercontent.com/assets/8058228/4475457/8805ed3e-496c-11e4-86a0-3a5a44f4d812.png)
![icon_512x512_c 2x](https://cloud.githubusercontent.com/assets/8058228/4475458/880c8f72-496c-11e4-826b-ddb29b68a030.png)

**Comment by ManfredKarrer** [comment]
@raulalgo: Interesting variants! 
The first I like most, in the others the ribbon between the 2 connected dots looks a bit strange as it is not really the same width. That is not so obvious in the normal version as well in the first version. The long shadow version looks nice as well, but I am not sure if it is good to follow that style which is very modern today, but maybe too much a temporary fashion?
The shadow at the frame in the first I like as well. 
I am not sure how the shadows inside the logo work when the logo is in small sizes (16, 32 px). Maybe we can use those only if the size is larger. It is common that icons change details with size.
Also the relation of the logo size to the frame-size might change so the logo itself has more weight (at least in smaller sizes).  
What do you think?

**Comment by raulalgo** [comment]
I'm not really sure what you mean by the ribbon.

As for the proportion between the logo and the frame I'll look for every size of the icon for the right proportion. That weight is carried so much by the logo size as for the extra negative space around it. Generally lighter graphics with that bit of extra space around end up being the most effective.

**Comment by ManfredKarrer** [comment]
@raulalgo: With the ribbon I meant the space between the 2 logo elements (connected points), which is nearly even spaced but not exactly. I see a kind of ribbon (or belt, band, tape) especially in the versions where logo is white. As that ribbon get more focus in the last 2 versions, for me the changing width of the ribbon becomes more obvious.

**Comment by raulalgo** [comment]
OK I see. The thing is that tin order to give that embossed look, the shadow is sitting on top of the adjacent space rather than the logo itself. It's happening in both blue and white versions but I can fix it.

**Comment by raulalgo** [comment]
I fixed the width changing ribbon and removed the shadow in the smaller sizes. If you are ok with it I can add these icons to the repository
![icon_512x512](https://cloud.githubusercontent.com/assets/8058228/4487891/29c04d98-4a05-11e4-9f9f-54ab3a1b4569.png)

**Comment by cbeams** [comment]
Hi Raul, go ahead and try committing them to the `native` branch. Thanks.

**Comment by raulalgo** [comment]
cool! a Straight commit is fine or are you following doing branching/pull request?

**Comment by cbeams** [comment]
You can commit this directly against the `native` branch if you like, but a pull request against that branch would be fine too.

**Comment by raulalgo** [comment]
Just commited it. I've been looking around where in the project to place the icons but could not find it. I've just added a folder at /src/main/resources/images/icon_set with all the variants. If there is anyone missing please let me know. I'll also be adding the source .sketch file in the next commit.

**Comment by cbeams** [comment]
Oops, looks like you committed this to `master` instead of `native`. I've rolled back master but saved your commits and will push them up shortly.

**Comment by raulalgo** [comment]
Damm, that's true. Sorry, I'm gifted for this kind of messes in git.

**Comment by cbeams** [comment]
No worries, I'll have something pushed up in a second and we can go from there.

**Comment by cbeams** [comment]
Raul, I've created #103 as a pull request containing your changes, but let's keep the comments here just so that it's all in one place. Take a look at the commit comment in 99101ab.

A few questions / thoughts:
1. Any reason that the 64px variant wasn't included?
2. You'll notice that the images have been moved into `src/deploy/package/shortcut-*` (these paths didn't exist on `master`, but they do exist on `native`).
3. I would say the icons look better in the dock and task switcher, but there's still quite a bit of whitespace. Seems like the bitsquare icon could fill the available real estate better. What do you think?

![screen shot 2014-10-02 at 10 52 51 am](https://cloud.githubusercontent.com/assets/301810/4488722/8f93083e-4a11-11e4-981b-98602faaad57.png)

With regard to 3., we can leave it as-is for now—as mentioned, this doesn't need to be perfect at this point, just presentable. But wanted to get your thoughts about that in any case.

Anyway, please review what you see in the pull request. I'll merge it if everything looks good to you.

Note that because I rolled back the earlier commits, you'll want to reset your `master` branch using the following commands:

```
git fetch origin
git checkout master
git reset --hard origin/master
```

And to actually build the OS X .app, .dmg and .pkg files with your new icon set, make sure you have the latest version of Java installed:

```
$ java -version
java version "1.8.0_20"
```

If yours does not read `1.8.0_20`, then you'll need to [download and install the latest JDK](http://www.oracle.com/technetwork/java/javase/downloads/jdk8-downloads-2133151.html) for Mac OS X x64.

Then do the following:

```
git checkout icons
./gradlew build
open ./build/distributions/Bitsquare.app
```

**Comment by raulalgo** [comment]
First, answering your questions:
1. It might be missing from the icon template I've used. I'll add it right now
2. I'm usually more in favor of balancing content and negative space. I don't like the versions in which the logo takes too much space from the white frame we are putting it into. For smaller versions, I've enlarged it a bit, but in general I think that extra space is positive for the overall look of the icon. However, I can do some tests with a larger logo and see how it looks.

As for the built. I've run everything. The built seems to have been succesfull, no errors prompted, but Bitsquare.app is missing. Any ideas of what might have been wrong?

**Comment by btcll** [comment]
I really like the icon, especially how it looks in the dock. 
Something to keep in mind:
If this needs to be running all the time that someone has an offer open then an unobtrusive dock icon is important. Some users might also be at work or in a public area so advertising they're running Bitsquare might not be something they want to do to loudly. The icon size on the white is a nice balance for me. I think with a red notification number when there are notifications in the app will be very noticable on that white background! Just my two cents :) Personally I would remove the dock icon on my system and just use the icon in the top task bar :)

**Comment by cbeams** [comment]
> It might be missing from the icon template I've used. I'll add it right now

Thanks. You can commit that against the `icons` branch and push it up. That way, it will be added to the pull request at #103. By the way, please follow these [seven rules for writing git commit messages](http://chris.beams.io/posts/git-commit/#seven-rules).

> I'm usually more in favor of balancing content and negative space. I don't like the versions in which the logo takes too much space from the white frame we are putting it into. For smaller versions, I've enlarged it a bit, but in general I think that extra space is positive for the overall look of the icon. However, I can do some tests with a larger logo and see how it looks.

Cool. I'm primarily concerned with following convention, such that we don't look out of place, particularly on OS X, which I assume most of our early adopters will be using. At this point, when I command-tab to switch applications, the bitsquare icon looks odd compared to the rest (I wish I could take a screenshot of the task switcher to demonstrate, but I don't know how to do this).

> As for the build. I've run everything. The built seems to have been succesfull, no errors prompted, but Bitsquare.app is missing. Any ideas of what might have been wrong?

Hm. If you've run `./gradlew build` against the `icons` branch, you should have the following files on your filesystem:

```
$ ls -1 build/distributions/
Bitsquare-0.1.0-SNAPSHOT.dmg
Bitsquare-0.1.0-SNAPSHOT.pkg
Bitsquare.app
```

At which point you should be able to run

```
open build/distributions/Bitsquare.app
```

**Comment by raulalgo** [comment]
None of them shows up. This is what I'm doing

```

➜  20 repo git:(icons) ./gradlew build             
:compileJava UP-TO-DATE
:processResources UP-TO-DATE
:cssToBin SKIPPED
:classes UP-TO-DATE
:jar
:jfxJar
:jfxSignJar SKIPPED
:jfxCopyLibs
:compilePackageJava UP-TO-DATE
:processPackageResources UP-TO-DATE
:packageClasses UP-TO-DATE
:jfxDeploy
:assemble
:compileTestJava UP-TO-DATE
:processTestResources UP-TO-DATE
:testClasses UP-TO-DATE
:test UP-TO-DATE
:check UP-TO-DATE
:build

BUILD SUCCESSFUL

Total time: 7.258 secs
➜  20 repo git:(icons) ls -1 ./build/distributions
➜  20 repo git:(icons) 



```

**Comment by cbeams** [comment]
That's strange. Just to make things simple, I've uploaded the .dmg that I built on my system here, so you can just try it out without any further hassle:

https://github.com/bitsquare/bitsquare/releases/download/v0.1.0-SNAPSHOT/Bitsquare-0.1.0-SNAPSHOT.dmg

However, it would be great if you'd run the following command and paste the console output here (or better yet, into a gist, just so we don't spam this thread with hundreds of lines of text).

```
./gradlew jfxDeploy --info
```

Seeing the output of this will help me diagnose why the files aren't getting created on your machine.

Thanks!

**Comment by raulalgo** [comment]
Here it is

https://gist.github.com/raulalgo/acc07b3baec49284d8eb

**Comment by raulalgo** [comment]
Ok, I've installed the .dmg and I have a couple of thoughts.

First, I don't feel white background is a good choice for us. It looks nice on the dock but it gets completely lost when seen on finder. I would like to give it a try to blue background, or at least do an alternative version with a light grey rather than pure white.

Also, I feel the oddity you feel at the task switcher has more to do with the fact that icon background goes all the way to the limit of icon dimensions, while most OSX icons are leaving transparent space around. I'll reduce that background section to see if it works better.

**Comment by cbeams** [comment]
> Here it is https://gist.github.com/raulalgo/acc07b3baec49284d8eb

Thanks. I've just committed something that may fix the problem. Try a `git pull` against the `icons` branch and try again.

**Comment by raulalgo** [comment]
Fails again

https://gist.github.com/raulalgo/3eda40a71f931c9c22d0

sorry for all the inconvenience u_u

**Comment by cbeams** [comment]
> Fails again https://gist.github.com/raulalgo/3eda40a71f931c9c22d0. sorry for all the inconvenience u_u

No worries. Thanks for trying to sort it out. I think it has to do with whitespace in your paths (though this is never usually a problem). This is just an educated guess based on output like this:

```
* What went wrong:
Execution failed for task ':jfxDeploy'.
> Unable to delete file: /Users/raulalgo/Documents/10 trabajos/2014 09 13 bitsquare/20 repo/build/javapackager-app/20 repo-0.1.0-SNAPSHOT.jar
```

So as a workaround, could you do a fresh clone into `/tmp`, e.g.:

```
cd /tmp
git clone https://github.com/bitsquare/bitsquare.git
cd bitsquare
./gradlew jfxDeploy --info
```

and see what happens?

**Comment by cbeams** [comment]
And Raul, in any case, the additional experimentation with icons that you mention above is quite welcome, thanks. Feel free to commit and push against the `icons` branch, and we can review from there.

**Comment by raulalgo** [comment]
Yes! It worked

**Comment by ManfredKarrer** [comment]
@raulalgo, @cbeams, @btcll: I have not followed all in details as I am busy preparing for todays presentation... But looks all good for me all and I am open for adjusting and refinements. In the App settings or config file we might add special handling like not showing the icon if a user want to hide the app, but that i would treat low prio for now. We can add a issue for it...

**Comment by cbeams** [comment]
Closed with bfa5855.

**Comment by ManfredKarrer** [comment]
I updated the icons. Removed shadow form icon, but let it aound border. We dont use the border shadow in the small sizes (32, 16 px) as it gets too blurry.
